storage/range_gc_queue.go

```diff
@@ -60,7 +60,7 @@ func (q *rangeGCQueue) needsLeaderLease() bool {
 // and if so at what priority. Currently all inactive ranges are
 // considered for possible GC at equal priority.
 func (q *rangeGCQueue) shouldQueue(now proto.Timestamp, rng *Range) (bool, float64) {
-	if _, expired := rng.HasLeaderLease(now); !expired {
+	if held, expired := rng.HasLeaderLease(now); held && !expired {
```

Comment:
I think the test could still be useful. It is a little redundant with `TestReplicateAddAndRemove`, but that test is doing a lot and it can be good to have a more focused test. One convention we've been following is to put tests that are in the `storage_test` package in files named `client_*_test.go` (e.g. `client_split_test.go` vs `split_queue_test.go`). Rename the file to client_range_gc_test.go and LGTM.